src/app/product-managment/create-product/create-product.component.ts

```diff
@@ -10,60 +12,106 @@ import { ProductModel } from './product.model';
-      .map(c => c.category);
-    // saveModel.categories = this.productFormGroup.controls.categories.value;
-    saveModel.tags = this.productFormGroup.controls.tags.value;
-    this.myService.saveProduct(saveModel).subscribe();
-    }
+  public pickCategory(selectedCategory: number): void {
+    const categories = this.productFormGroup.controls.categories as FormArray;
+    categories.push(this.formBuilder.control(this.categories[selectedCategory].category));
+    this.myService.addCategory(selectedCategory);
+    this.myService.getCategories().subscribe(c => {
+      this.categories = c;
+    })
   }
 
-  public addtag(): void {
+  public removeCategory(index): void {
+    const categories = this.productFormGroup.controls.categories as FormArray;
+    categories.push(this.formBuilder.control(this.categories[index].category));
+    this.myService.removeCategory(index);
+    this.myService.getCategories().subscribe(c => {
+      this.categories = c;
+    })
+  }
+  public removeTag(index: number): void{
     const tags = this.productFormGroup.controls.tags as FormArray;
-    tags.push(this.formBuilder.control(''));
+    tags.removeAt(index);
+  }
+
+  public submitProduct(): void {
+    if (this.productFormGroup.valid) {
+      const saveModel = new ProductModel();
+      const name: AbstractControl = this.productFormGroup.controls.name
+
+      this.getFirstLetterToUpperCase(name);
+      saveModel.name = this.productFormGroup.controls.name.value;
+      saveModel.description = this.productFormGroup.controls.description.value;
+      saveModel.categories = this.categories
+                                 .filter(c => c.selected)
+                                 .map(c => c.category);
+      saveModel.tags = this.productFormGroup.controls.tags.value;
+
+      this.myService
+        .saveProduct(saveModel)
+        .subscribe(() => {
+          this.successPopUp();
+          this.resetForm();
+        });
+    }
   }
 
-  pickCategory(selectedCategory: number): void {
-    this.categories[selectedCategory].selected = true;
-    const tags = this.productFormGroup.controls.categories as FormArray;
-    tags.push(this.formBuilder.control(this.categories[selectedCategory].category));
+  public resetForm(): void {
+    this.productFormGroup = this.formBuilder.group({
+      name: ['', [Validators.required, Validators.minLength(3), Validators.maxLength(12)]],
+      tags: this.formBuilder.array([]),
+      categories: this.formBuilder.array([], Validators.required),
+      description: ['', Validators.required],
+    });
+
+    this.categories.map((c)  =>  {
+      if(c.selected === true) c.selected = false
+      // random element get true and i dont know why
+      this.resetedArray.push(c);
+    });
+    this.categories = this.resetedArray;
+    this.resetedArray = [];
   }
 
-  removeCategory(index: number): void {
-    this.categories[index].selected = false;
+  private getFirstLetterToUpperCase (name: AbstractControl): void  {
```

Comment:
Funtion that does not return a value should not be named get. You can name it for example patchNameControll or patchNameControllWithFirstLetterUppercase